firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformance.java

```diff
@@ -132,13 +133,17 @@ public static FirebasePerformance getInstance() {
   // to false if it's been force disabled or it is set to null if neither.
   @Nullable private Boolean mPerformanceCollectionForceEnabledState = null;
 
-  /** This constructor is invoked from {@link com.google.firebase.perf.FirebasePerfRegistrar}. */
-  FirebasePerformance(
-      FirebaseApp firebaseApp,
-      Provider<RemoteConfigComponent> firebaseRemoteConfigProvider,
-      FirebaseInstallationsApi firebaseInstallationsApi,
-      Provider<TransportFactory> transportFactoryProvider) {
+  FirebaseApp firebaseApp;
+  Provider<RemoteConfigComponent> firebaseRemoteConfigProvider;
+  FirebaseInstallationsApi firebaseInstallationsApi;
+  Provider<TransportFactory> transportFactoryProvider;
```

Comment:
Can these be final?

Reply:
Potentially also private

Reply:
Made it private. Will move it to final as well soon!

Reply:
Just checking for the "final" bit - is it something coming in a different PR?

Reply:
This is done now.